solidity/random-beacon/README.adoc

```diff
@@ -177,103 +166,122 @@ When the hard timeout is reached, anyone can notify about this fact by calling
-
-[NOTE]
-For example, if `hash(new_group_pubkey) % group_size = 62`, `group_size = 64`,
-group members are becoming eligible in the following order:
-`62, 63, 64, 1, 2, 3, 4, 5, 6, 7, 8, 9, ..., 61`. 
-
-The transaction submitting relay entry is not reimbursable and implementation
-ensures the gas cost of this transaction is as low as possible, below 200k gas
-when no callback is executed.
-
-Everyone is eligible to submit relay entry at any time but off-chain clients are
-expected to agree and follow the following order to minimize the gas cost and
-distribute costs: the first group member eligible to submit the result is
-`new_entry % group_size`; then, if the selected member does not provide an entry
-within the eligibility period, `(new_entry % group_size) + 1` and so on.
-
-If some group members are notoriously ignoring their duty, the group can vote on
-<<inactivity,inactivity>> notification for these operators.
-
-T rewards will be distributed continuously to all operators registered in the beacon
-sortition pool, excluding operators who were marked as ineligible for rewards
-due to failing the heartbeat.
+to generate a key, and the time it takes to submit the result.
+When DKG timeout is exceeded, anyone can call `RandomBeacon.notifyDkgTimeout()`.
+This function unlocks the sortition pool and clears up DKG data but no slashing
+for DKG timeout is executed and no one is marked as ineligible for rewards.
 
 [[inactivity]]
 === Inactivity notification
 
 Off-chain clients are free to execute any heartbeat protocol they want to ensure
-group members are alive and nodes are operating properly.
+group member key material is still available and nodes are operating properly.
 
 [TIP]
 One example of a heartbeat protocol is signing some piece of information every
-nth blocks and first making sure the information cannot be used for
-`RandomBeacon.reportUnauthorizedSigning()`, that is, the signed information can
-not become `msg.sender` for `reportUnauthorizedSigning` call.
+nth block and making sure this piece of information cannot be used for
+`RandomBeacon.reportUnauthorizedSigning()`. Specifically, the signed piece of
+information can not become `msg.sender` for `reportUnauthorizedSigning` call.
 
-Group members can agree upon members who are permanently inactive and issue an
-operator inactivity claim. If the required threshold of group members signed
+Group members can agree to punish members who are permanently inactive and issue
+an operator inactivity claim. If the required threshold of group members signed
 the operator inactivity claim, they can submit it to
-`RandomBeacon.notifyOperatorInactivity(Inactivity.Claim calldata claim, uint256 nonce)`
+`RandomBeacon.notifyOperatorInactivity(Inactivity.Claim calldata claim, uint256 nonce, int32[] calldata groupMembers)`
 function and have the group members who are inactive excluded from
 the sortition pool rewards for a governable time period.
```

Comment:
Could we link to Rewards section here?

Reply:
1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322